Hyper/MarchingCubes/ChunkWorker.cs

```diff
@@ -0,0 +1,114 @@
+
+namespace Hyper
+{
+    internal class ChunkWorker
+    {
+        private ConcurrentDictionary<Vector3i, Guid> _neededChunkPositions;
+        private ConcurrentDictionary<Vector3i, Guid> _existingChunkPositions;
+        private ConcurrentDictionary<Guid, Chunk> _existingChunks;
+        private CancellationTokenSource _cancellationTokenSource;
+        private ScalarFieldGenerator _scalarFieldGenerator;
+        private ChunkFactory _chunkFactory;
+        private SimulationManager<NarrowPhaseCallbacks, PoseIntegratorCallbacks> _simulationManager;
+
+        public ChunkWorker(ConcurrentDictionary<Guid, Chunk> existingChunks, ScalarFieldGenerator scalarFieldGenerator, 
+            SimulationManager<NarrowPhaseCallbacks, PoseIntegratorCallbacks> simulationManager)
+        {
+            _cancellationTokenSource = new CancellationTokenSource();
+            _neededChunkPositions = new ConcurrentDictionary<Vector3i, Guid>();
+            _neededChunkPositions.TryAdd(new Vector3i(0, 0, 0), Guid.NewGuid()); // initial chunk...
+            _existingChunkPositions = new ConcurrentDictionary<Vector3i, Guid>();
+            _existingChunks = existingChunks;
+            _scalarFieldGenerator = scalarFieldGenerator;
+            _chunkFactory = new ChunkFactory(_scalarFieldGenerator);
+            _simulationManager = simulationManager;
+        }
+        public void StartProcessing()
+        {
+            var processingThread = new Thread(ProcessItems);
+            processingThread.IsBackground = true;
+            processingThread.Start();
+        }
+
+        public void StopProcessing()
+        {
+            _cancellationTokenSource.Cancel();
+        }
+
+        private void ProcessItems()
+        {
+            while (!_cancellationTokenSource.IsCancellationRequested)
+            {
+                Parallel.ForEach(_neededChunkPositions, kvp =>
+                {
+                    if (!_existingChunkPositions.ContainsKey(kvp.Key))
+                    {
+                        _existingChunkPositions.TryAdd(kvp.Key, kvp.Value);
+                        var chunk = _chunkFactory.GenerateChunk(kvp.Key);
+                        chunk.CreateCollisionSurface(_simulationManager.Simulation, _simulationManager.BufferPool);
+                        _existingChunks.TryAdd(kvp.Value, chunk);
+                    }
+                });
+                Parallel.ForEach(_existingChunkPositions, kvp =>
+                {
+                    if (!_neededChunkPositions.ContainsKey(kvp.Key))
+                    {
+                        _existingChunkPositions.TryRemove(kvp.Key, out _);
+                        _existingChunks.TryRemove(kvp.Value, out var chunk);
+                        // multithreaded removal might need a mutex?
+                        // chunk?.DisposeCollisionSurface(_simulationManager.Simulation, _simulationManager.BufferPool);
+                    }
+                });
+                Thread.Sleep(100);
```

Comment:
Why do we need a sleep? Maybe call this on every update frame or do something else to avoid wasting a thread